### PR TITLE
Speed-up polygon soup orientation

### PR DESCRIPTION
**Pull request by sloriot**
Replace `map< pair<int,int>, flat_set<int> >` by `vector< flat_map<int, flat_set<int> >`  for the edge map.
The speed up is huge because we use a lot the edge map for navigation and the new data set is efficient because in regular cases, there are no isolated vertices.

### CODE REVIEW

**Comment by lrineau** [comment]
Only `master`?

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup.h [hunk 1]
```diff
@@ -108,8 +110,8 @@ struct Polygon_soup_orienter
   {
     typedef std::pair<V_ID,P_ID> VID_and_PID;
     if ( is_edge_marked(src,tgt,marked_edges) ) return VID_and_PID(src,300612);
-    Edge_map_iterator em_it=edges.find(V_ID_pair(tgt, src));
-    if ( em_it==edges.end() ) return VID_and_PID(src,300612);// the vertex is on the border
+    Edge_map_iterator em_it=edges[tgt].find(src);
+    if ( em_it==edges[tgt].end() ) return VID_and_PID(src,300612);// the vertex is on the border
```
Comment (lrineau) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup.h]: If you want to use magic number, use sound ones, like `-1`, but not arbitrary integers!

**Comment by maxGimeno** [comment]
Travis reports several errors:
https://travis-ci.org/CGAL/cgal/jobs/516122021#L3064
https://travis-ci.org/CGAL/cgal/jobs/516122029#L2999
https://travis-ci.org/CGAL/cgal/jobs/516122039#L2809

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup.h (hunk 1 above)
Comment (sloriot) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup.h]: The change does not introduce this number it was there before.

**Comment by maxGimeno** [comment]
Successfully tested in  https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-16.shtml

**Comment by lrineau** [comment]
This PR caused a compilation error in `releases/CGAL-4.14-branch`:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14.1-I-166/Polyhedron_Demo/TestReport_lrineau_Fedora-Release.gz

**Comment by lrineau** [comment]
Fixed in commit 89c7edec558.

**Pull request merged**